src/main/java/org/advent/daySeven/Hand.java

```diff
@@ -0,0 +1,174 @@
+        for (Character character : mapOfLettersAndHowMany.keySet()) {
+            for (int i = 0; i < mapOfLettersAndHowMany.get(character); i++) {
+                listOfCards.add(character);
+            }
+        }
+        return listOfCards;
+    }
+
+    private Type includeJokers(Type type) {
+
+        if (type == Type.TWO_PAIR
+                && anyEqualsJ(relevantCards)) {
+            return Type.FOUR_OF_A_KIND;
+        }
+
+        if (type == Type.TWO_PAIR
+                && anyEqualsJ(restOfHand)) {
+            return Type.FULL_HOUSE;
+        }
+
+        if (type == Type.FULL_HOUSE
+                && anyEqualsJ(relevantCards)) {
+            return Type.FIVE_OF_A_KIND;
+        }
+
+        if (type == Type.FOUR_OF_A_KIND
+                && (anyEqualsJ(restOfHand) || allEqualsJ(relevantCards))) {
+            return Type.FIVE_OF_A_KIND;
+        }
+
+        if (type == Type.THREE_OF_A_KIND
+                && (allEqualsJ(relevantCards) || anyEqualsJ(restOfHand))) {
+            return Type.FOUR_OF_A_KIND;
+        }
+
+        if (type == Type.ONE_PAIR
+                && (allEqualsJ(relevantCards) || anyEqualsJ(restOfHand))) {
+            return Type.THREE_OF_A_KIND;
+        }
+
+        if (type == Type.HIGH_CARD
+                && anyEqualsJ(restOfHand)) {
+            return Type.ONE_PAIR;
+        }
+
+        return type;
+    }
+
+    private boolean anyEqualsJ(List<Character> chars) {
+        return chars.stream().anyMatch(ch -> ch.equals('J'));
+    }
+
+    private boolean allEqualsJ(List<Character> chars) {
+        return chars.stream().allMatch(ch -> ch.equals('J'));
+    }
+
+    private Type typeOfHand(boolean isPartTwo) {
+
+        Type type = null;
+        if (!allCardsMatch() && isListSize(4)) {
+            type = Type.TWO_PAIR;
+        }
+
+        if (!allCardsMatch() && isListSize(5)) {
+            type = Type.FULL_HOUSE;
+        }
+
+        if (allCardsMatch() && isListSize(5)) {
+            type = Type.FIVE_OF_A_KIND;
+        }
+
+        if (allCardsMatch() && isListSize(4)) {
+            type = Type.FOUR_OF_A_KIND;
+        }
+
+        if (allCardsMatch() && isListSize(3)) {
+            type = Type.THREE_OF_A_KIND;
+        }
+
+        if (allCardsMatch() && isListSize(2)) {
+            type = Type.ONE_PAIR;
+        }
+
+        if (relevantCards.isEmpty()) {
+            type = Type.HIGH_CARD;
+        }
+
+        if (isPartTwo) {
+            type = includeJokers(type);
+        }
+
+        return type;
+    }
+
+    private boolean isListSize(int size) {
+        return relevantCards.size() == size;
+    }
+
+    private boolean allCardsMatch() {
+        return relevantCards.stream().allMatch(ch -> ch.equals(relevantCards.get(0)));
+    }
+
+    private String handOfString(String string) {
+        return string.substring(0, string.indexOf(" "));
+    }
+
+    private List<Character> findNonRelevantCardsInHand() {
```

Comment:
im pretty confused what relevant and non relevant cards are

Reply:
relevant cards are cards that are part of whatever type, i.e. three of a kind 555AQ, relevantCards = 555, nonReleventCards = AQ. Changed name to "typeCards" and "NonTypeCards", any suggestions?